dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -99,6 +100,51 @@ public static class CommandLineParser
                 "--http-port",
                 "Specifies the port on which to enable HTTP services");
 
+            var httpPortRangeOption = new Option<PortRange>(
+                "--http-port-range",
+                parseArgument: result =>
+                {
+                    if (result.Parent.Parent.Children.Any(c => c.Symbol == httpPortOption))
+                    {
+                        result.ErrorMessage = $"Cannot specify both {httpPortOption.Name} and {result.Symbol.Name} together";
+                        return null;
+                    }
+
+                    var pr = new PortRange();
+
+                    var source = result.Tokens[0].Value;
+
+                    if (string.IsNullOrWhiteSpace(source))
+                    {
+                        result.ErrorMessage = "Must specify a port range";
+                        return null;
+                    }
+
+                    var parts = source.Split(new[] {"-"}, StringSplitOptions.RemoveEmptyEntries);
+
+                    if (parts.Length != 2)
+                    {
+                        result.ErrorMessage = "Must specify a port range";
+                        return null;
+                    }
```

Comment:
Can this be cleaned up so the error cases aren't scattered around the method? This one should be able to be consolidate into the one below, which is more informative because it also tells people the form that the argument needs to take.